elastic/kibana
# [Logs UI] Versioning for Logs related APIs

## 📓  Summary

Closes #157324 
Closes #159275 
Closes #159303 

This work converts the APIs related to the Logs UI feature into versioned APIs using the new [Kibana versioned router](https://docs.elastic.dev/kibana-dev-docs/versioning-http-apis#4-adhere-to-the-http-versioning-specification).

The converted APIs are the following, where each endpoint now is set to version `1`:
- [log_views](https://github.com/elastic/kibana/tree/main/x-pack/plugins/infra/server/routes/log_views)
- [log_entries](https://github.com/elastic/kibana/tree/main/x-pack/plugins/infra/server/routes/log_entries)
- [log_analysis](https://github.com/elastic/kibana/tree/main/x-pack/plugins/infra/server/routes/log_analysis)
- [log_alerts](https://github.com/elastic/kibana/tree/main/x-pack/plugins/infra/server/routes/log_alerts)

The PR also includes moving the interfaces and runtime types relatives to each endpoint's group into the recommended practices for [versioning interfaces](https://docs.elastic.dev/kibana-dev-docs/versioning-interfaces).

## 🧪 Testing

- Navigate to the Logs UI settings page and verify the log view is correctly retrieved and can be successfully updated.
- Navigate to the Logs stream page and verify the stream entries are retrieved and rendered.
- Navigate to the Anomalies and Categories pages page and verify the anomalies entries are retrieved and rendered correctly.
- Create a Log threshold alert and verify the chart preview data are correctly retrieved and shown.
